component/http/middleware.go

```diff
@@ -153,6 +159,72 @@ func ignore(ignoreRoutes []string, url string) bool {
 	return false
 }
 
+func parseAcceptEncoding(header string) (string, error) {
+	if header == anythingHeader {
+		return anythingHeader, nil
+	}
+
+	weighted := make(map[float64]string)
+
+	algorithms := strings.Split(header, ",")
+	for _, a := range algorithms {
+		algAndWeight := strings.Split(a, ";")
+		algorithm := strings.TrimSpace(algAndWeight[0])
+
+		if notSupportedCompression(algorithm) {
+			continue
+		}
+
+		if len(algAndWeight) != 2 {
+			addWithWeight(weighted, 0.0, algorithm)
+			continue
+		}
+
+		qWeight := algAndWeight[1]
+		weight, err := parseWeight(qWeight)
+		if err != nil {
+			addWithWeight(weighted, 0.0, algorithm)
+			continue
+		}
+
+		addWithWeight(weighted, weight, algorithm)
+	}
+
+	return selectByWeight(weighted)
+}
+
+func addWithWeight(mapWeighted map[float64]string, weight float64, algorithm string) {
+	if _, ok := mapWeighted[weight]; !ok {
+		mapWeighted[weight] = algorithm
+	}
+}
+
+func notSupportedCompression(algorithm string) bool {
+	return gzipHeader != algorithm && deflateHeader != algorithm && anythingHeader != algorithm
+}
+
+func parseWeight(qStr string) (float64, error) {
```

Comment:
I think that we could add one more check, as the valid [Quality Values](https://developer.mozilla.org/en-US/docs/Glossary/Quality_values) are in the [0, 1] range.

Reply:
Yes, we can do it.
There are two ways to solve this
```
if parsedWeight < 0.0 { return 0.0}
if parsedWeight > 1.0 { return 1.0}
```
or 
```
if parsedWeight < 0.0 || parsedWeight > 1.0 { return err }
```

WDYT is better?

Reply:
@azarakovskiy I'm not sure if the RFCs mention what should be the correct behavior, but I'd lean over to the second approach; return an error from 'parseWeight' and assign a zero weight.

Reply:
RFC: `When not present, the default value is 1.`
I'd argue that a wrong `q` is similar to `not present`

Reply:
(For reference, it's [RFC 2616 Section 14.1](https://tools.ietf.org/html/rfc2616#section-14.1).

Let's go with this.